src/containers/other/NFTView/NFTView.tsx

```diff
@@ -310,6 +350,34 @@ const NFTView: React.FC<NFTViewProps> = ({ onTransfer }) => {
 		};
 	}, []);
 
+	useEffect(() => {
+		(async () => {
+			const isAbleToSpendTokens = await (
+				await sdk.instance.getZAuctionInstanceForDomain(domainId)
+			).getZAuctionSpendAllowance(await signer?.getAddress()!);
+
+			if (isAbleToSpendTokens) {
+				const zauctionBuyNowPrice = (await (
+					await sdk.instance.getZAuctionInstanceForDomain(domainId)
+				).getBuyNowPrice(domainId, signer!)) as BigNumber;
+
+				if (!zauctionBuyNowPrice.isZero())
+					setBuyNowPrice(zauctionBuyNowPrice.toString());
+			} else {
+				await (await sdk.instance.getZAuctionInstanceForDomain(domainId))
```

Comment:
As mentioned in other PR can we please simplify this or move it to zns sdk provider

Reply:
To add to this, I think using `sdk.instance.zauction` will give you the same result, rather than calling `getZAuctionInstanceForDomain(...)`